rtl/eclrtl/rtlrecord.cpp

```diff
@@ -303,6 +303,34 @@ size32_t RtlRecord::getMinRecordSize() const
     return minSize;
 }
 
+size32_t RtlRecord::deserialize(ARowBuilder & rowBuilder, IRowDeserializerSource & in) const
+{
+    size32_t offset = 0;
+    for (unsigned i = 0; i < numVarFields; i++)
+    {
+        unsigned fieldIndex = variableFieldIds[i];
+        size32_t fixedSize = fixedOffsets[fieldIndex];
+        byte * self = rowBuilder.ensureCapacity(offset + fixedSize, "");
+        in.read(fixedSize, self + offset);
```

Comment:
Note - also deserializes any fixed-size fields that precede this one (worth a comment)

Reply:
Could potentially be coded as a readahead then a transform - would know size up-front then (and would handle link-counted rows).

Reply:
Also, do we ever really want to call deserialize?